crates/core/src/lib.rs

```diff
@@ -25,6 +25,16 @@ pub type ActorLinks = Vec<LinkDefinition>;
 pub type ClusterIssuerKey = String;
 pub type ClusterIssuers = Vec<ClusterIssuerKey>;
 
+/// Any string that represents a link between two components on the lattice,
+/// possibly a `ComponentId` or or a target group
+pub type LinkName = String;
```

Comment:
The link name serves the same purpose as it did before, simply the "name" on the InterfaceLinkDefinition. The "target" will be the part that's possibly a component ID or target group

Reply:
Ah OK, I'll change the comment here to be something like:

```
/// Name of an established link ([`InterfaceLinkDefinition`]) 
```